### PR TITLE
Show topics in search results

### PR DESCRIPTION
**Pull request by Reekamal**
https://trello.com/c/qHQ0v73M/101-show-topics-in-search-results

### CODE REVIEW

File: frontend/src/index.html [hunk 1]
```diff
@@ -126,7 +127,7 @@
         </div>
         <div id="hits" class="overflow-x-hidden">
           <template id="item-template">
-            <article class="flex flex-col mb-4 p-1 w-full bg-white rounded-sm shadow-md sm:flex-row sm:p-4" data-file-id="{{objectID}}">
+            <article class="flex flex-col mb-4 p-1 w-full bg-white rounded-sm shadow-md sm:flex-row sm:p-4" x-data="{is_open:false}" data-file-id="{{objectID}}">
```
Comment (anantakrishna) [inline frontend/src/index.html:130]: Let's keep the directives content (JS expressions) formatted. I've tried to look for an automatic way, similar to what Prettier does for separate JS files, but could not find such a tool for Alpine yet. Therefore let's maintain proper formatting manually as of now.

I cannot explain all the formatting conventions here, but essentially I'd expect us to use the same formatting as Prettier enforces. You can paste your code into a separate JS file to see how Prettier would format it. Thus gradually you will get accustomed to the conventions.

File: frontend/src/index.html [hunk 2]
```diff
@@ -177,6 +178,7 @@ <h3 class="flex-grow break-words font-semibold" title="{{title}}">{{#helpers.hig
                     {{#_highlightResult.languages}}
                     <div title="Language spoken in the recording">{{{value}}}</div>
                     {{/_highlightResult.languages}}
+                    <a x-on:click="is_open=!is_open" href="#" class="flex flex-col flex-grow text-center">More...</a>
```
Comment (anantakrishna) [inline frontend/src/index.html:181]: - I'm sure this link should be shown only if `topicsReady === true`. 
- Can we make it a span/div instead of a dummy anchor? It may have `cursor-pointer` class to indicate that it's clickable.
- It doesn't seem to have to be a `flex` because it does not have children. Actually I don't think any of the provided classes are needed here.

File: frontend/src/index.html [hunk 3]
```diff
@@ -200,6 +202,11 @@ <h3 class="flex-grow break-words font-semibold" title="{{title}}">{{#helpers.hig
                     </a>
                   </div>
                 </div>
+                <div x-show="is_open" @click="text_open=false" class="mt-1">
+                  {{#_highlightResult.topics}}
+                  <h3 title="{{topics}}">{{#helpers.highlight}} { "attribute": "title" }{{/helpers.highlight}}>{{{value}}}</h3>
```
Comment (anantakrishna) [inline frontend/src/index.html:207]: - It shouldn't be `h3`. Use `p` instead.
- Topics are too large to use it in `title` attribute. Better to put some explanatory text like “Topics discussed in the recording” there.
- `{{#helpers.highlight}} { "attribute": "title" }{{/helpers.highlight}}` seem to be excessive. As soon as you have already opened `{{#_highlightResult.topics}}` section, `{{{value}}}` should be enough.
- There is a `>` symbol which seems to be a typo. Moreover, HTML linter warns about it: 
image

File: frontend/src/index.html [hunk 4]
```diff
@@ -200,6 +202,11 @@ <h3 class="flex-grow break-words font-semibold" title="{{title}}">{{#helpers.hig
                     </a>
                   </div>
                 </div>
+                <div x-show="is_open" @click="text_open=false" class="mt-1">
```
Comment (anantakrishna) [inline frontend/src/index.html:205]: - Don't close the topics on click. That will not be convenient. People would need to select and copy some text from it.
- Do we really need a wrapper `div` here? Can't we apply directives and classes directly to the nested element?

**Review by anantakrishna — CHANGES_REQUESTED** [review]

File: frontend/package-lock.json [hunk 5]
```diff
@@ -1,8 +1,13005 @@
 {
   "name": "frontend",
   "version": "0.1.0",
-  "lockfileVersion": 1,
+  "lockfileVersion": 2,
```
[the remark this answers is not in the document]
Comment (anantakrishna) [reply frontend/package-lock.json:4]: This is not done yet.

File: frontend/src/index.html (hunk 1 above)
Comment (anantakrishna) [reply frontend/src/index.html:129]: Also, let's follow JS naming conventions, i.e. `isOpen`.

File: frontend/src/index.html [hunk 6]
```diff
@@ -177,9 +177,13 @@ <h3 class="flex-grow break-words font-semibold" title="{{title}}">{{#helpers.hig
...
                     <div title="Language spoken in the recording">{{{value}}}</div>
                     {{/_highlightResult.languages}}
+                    <!--Show Topics-->
+                    {{#topicsReady}}
+                    <div x-on:click="is_open=true" class="cursor-pointer"> More... </div>
+                    {{/topicsReady}}
                   </div>
                   <div class="flex children:ml-1 children:pl-1 text-sm divide-x">
-                    <!-- Download -->
+                    <!-- Download  -->
```
Comment (anantakrishna) [inline frontend/src/index.html:186]: This change is irrelevant.

File: frontend/src/index.html [hunk 7]
```diff
@@ -200,6 +204,14 @@ <h3 class="flex-grow break-words font-semibold" title="{{title}}">{{#helpers.hig
                     </a>
                   </div>
                 </div>
+                <!--Topics discussed in the recording-->
+                {{#_highlightResult.topics}}
+                <div x-show="is_open" @click="text_open=false"><ul class="ml-7 mt-2 text-sm list-disc list-inside">
```
Comment (anantakrishna) [inline frontend/src/index.html:209]: What is `text_open=false` for? First, the property name does not match the `x-data`. Secondly, we don't need to hide the topics on click.

Comment (anantakrishna) [reply frontend/src/index.html:209]: Also, it seems there is some open-closing tag mismatch here, please check.

File: frontend/src/index.html [hunk 8]
```diff
@@ -200,6 +204,14 @@ <h3 class="flex-grow break-words font-semibold" title="{{title}}">{{#helpers.hig
                     </a>
                   </div>
                 </div>
+                <!--Topics discussed in the recording-->
+                {{#_highlightResult.topics}}
```
Comment (anantakrishna) [inline frontend/src/index.html:208]: Having different condition for displaying the topics in comparison to displaying the More link may lead to inconsistency. Let's use `#topicsReady` here also.

File: frontend/src/app.js [hunk 9]
```diff
@@ -3,6 +3,7 @@
  */
 
 import algoliasearch from 'algoliasearch/lite';
+import 'alpinejs';
```
Comment (anantakrishna) [inline frontend/src/app.js:6]: I believe you need to install the package (i.e. add it to the `package.json`) to be able to import it like this.

File: frontend/src/app.js [hunk 10]
```diff
@@ -12,7 +13,7 @@ import {
   pagination,
   panel,
   refinementList,
-  stats,
+  stats
```
Comment (anantakrishna) [inline frontend/src/app.js:16]: This change seems irrelevant.

File: frontend/src/app.js [hunk 11]
```diff
@@ -170,9 +171,10 @@ search.addWidgets([
       item: document.getElementById('item-template').innerHTML,
     },
     transformItems: (items) =>
-      items.map((item) => ({
+     items.map((item) => ({
```
Comment (anantakrishna) [inline frontend/src/app.js:174]: This change seems irrelevant.

File: frontend/src/index.html [hunk 12]
```diff
@@ -177,9 +177,13 @@ <h3 class="flex-grow break-words font-semibold" title="{{title}}">{{#helpers.hig
                     {{#_highlightResult.languages}}
                     <div title="Language spoken in the recording">{{{value}}}</div>
                     {{/_highlightResult.languages}}
+                    <!--Show Topics-->
+                    {{#topicsReady}}
+                    <div x-on:click="is_open=true" class="cursor-pointer"> More... </div>
```
Comment (anantakrishna) [inline frontend/src/index.html:182]: This link should toggle the topics (hide/show) rather than just showing it.

**Review by anantakrishna — CHANGES_REQUESTED** [review]

File: frontend/src/app.js (hunk 9 above)
Comment (Reekamal) [reply frontend/src/app.js]: There were 2 options of installing alpinejs, either via CDN or NPM.
I followed the instructions below:

NPM installation
Include the following in your script using import alpinejs.
npm i alpinejs

Comment (anantakrishna) [reply frontend/src/app.js]: Right. The point is that `npm i alpinejs` should add it to `package.json`, but it was not the case. Now after I merged the main branch into yours, it appeared in `package.json` because it was added to the project in some other task in parallel to yours. So never mind it as of now.

File: frontend/src/index.html (hunk 7 above)
Comment (Reekamal) [reply frontend/src/index.html]: I replaced @click="text_open=false" with @click="isOpen=!isOpen" after following JS naming convention. This allows your following request to be fullfilled :
This link should toggle the topics (hide/show) rather than just showing it.

Otherwise if I delete @click="isOpen=!isOpen" the topics  will just show when "More" link is clicked. I am little confused  because you said " Secondly, we don't need to hide the topics on click."

These are the following changes I made which allows the "More" link to toggle the topics (hide/show) when clicked:

                    {{#topicsReady}}
                    <div x-on:click="isOpen=!isOpen" class="cursor-pointer"> More... </div>
                    {{/topicsReady}}

                  {{#topicsReady}}
                <div x-show="isOpen" @click="isOpen=!isOpen"><ul class="ml-7 mt-2 text-sm list-disc list-inside">
                  {{#topicsArray}}
                  <li>{{{.}}}</li>
                  {{/topicsArray}}
                </div>
                {{/topicsReady}}

Comment (anantakrishna) [reply frontend/src/index.html]: I mean that we should not hide the topics by clicking *them*. Only the “More” link should control the topics visibility.

You don't need to post changes to the comment, just make new commits and push them, I will see the changes.
